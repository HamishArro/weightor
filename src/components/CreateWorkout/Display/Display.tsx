import React from 'react';
import {View, StyleSheet, Text, Button} from 'react-native';
import {DisplayProps} from '../types';

const styles = StyleSheet.create({
  displayContainer: {
    margin: 32,
    alignItems: 'center',
  },
});

function Display({exercises, addButtonText, addHandler}: DisplayProps) {
  return (
    <View style={styles.displayContainer} testID="display-container">
      {exercises.map(({name}) => (
        <Text testID={`exercise-${name}`} children={name} />
      ))}
      <Button testID="add-button" title={addButtonText} onPress={addHandler} />
    </View>
  );
}

export default Display;
