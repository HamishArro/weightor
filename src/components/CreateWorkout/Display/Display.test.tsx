import * as React from 'react';
import {describe, it, expect, beforeEach, jest} from '@jest/globals';
import {fireEvent, render, screen} from '@testing-library/react-native';
import Display from './Display';

const cardioExercise = {
  name: 'cardio',
  musclesUsed: ['legs'],
  workoutEffort: 50,
  duration: 300,
};

const props = {
  exercises: [cardioExercise],
  addHandler: jest.fn(),
  addButtonText: 'Add',
};

describe('display tests', () => {
  beforeEach(() => {
    render(<Display {...props} />);
  });

  it('calls add handler when button is pressed', () => {
    fireEvent.press(screen.getByTestId('add-button'));

    expect(props.addHandler).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
