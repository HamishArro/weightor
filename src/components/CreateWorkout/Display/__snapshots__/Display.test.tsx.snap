// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display tests matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "margin": 32,
    }
  }
  testID="display-container"
>
  <Text
    testID="exercise-cardio"
  >
    cardio
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="add-button"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add
      </Text>
    </View>
  </View>
</View>
`;
